dockers/tpu-tests/base_tpuvm.libsonnet

```diff
@@ -0,0 +1,146 @@
+local utils = import 'templates/utils.libsonnet';
+local volumes = import 'templates/volumes.libsonnet';
+
+{
+  BaseTpuVmMixin:: {
+    local config = self,
+    local cleanupHook = {
+      preStop: {
+        exec: {
+          command: [
+            'bash',
+            '/scripts/cleanup.sh',
+          ],
+        },
+      },
+    },
+
+    publisherImage: null,
+    volumeMap+: {
+      scripts: volumes.MemoryVolumeSpec {
+        name: 'scripts',
+        mountPath: '/scripts',
+      },
+    },
+
+    testName+: '-1vm',
+
+    tpuSettings+: {
+      local tpuSettings = self,
+
+      softwareVersion: if config.accelerator.replicas == 1 then
```

Comment:
yours can probably always be `v2-nightly` because I'd recommend avoiding pod testing to start

`v2-nightly` will be the unstable TPUVM verison. You could use `v2-alpha` to get the more stable version which is probably good since you'll want to isolate the failures that came from your code, not from the changing TPUVM image